src/api/spec/models/bs_request_spec.rb

```diff
@@ -156,6 +156,42 @@
         expect(request.history_elements.first.type).to eq('HistoryElement::RequestDeleted')
       end
     end
+
+    context 'final state declined cannot be changed ' do
+      let(:request) do
+        create(:declined_bs_request,
+               target_package: target_package,
+               target_project: target_project,
+               source_package: source_package,
+               source_project: source_project)
+      end
+      let(:user) { create(:confirmed_user) }
+
+      before do
+        User.current = user
+      end
+
+      it { expect { request.change_state(newstate: 'review') }.to raise_error(PostRequestNoPermission) }
+    end
+
+    context 'final state accepted cannot be changed' do
+      let!(:request) do
+        create(:bs_request_with_submit_action,
+               target_project: target_project.name,
+               source_project: source_package.project.name,
+               source_package: source_package.name)
+      end
+      let!(:relationship_project_user) { create(:relationship_project_user, project: target_project) }
+      let(:user) { relationship_project_user.user }
+      before do
+        User.current = user
+        request.state = 'accepted'
+        request.save
+      end
+
+      it { expect(request.state).to eq(:accepted) }
```

Comment:
I actually expected it to fail, we shouldn't be able to move a state from final state like ```:accepted``` to something different, I just wanted to use it to start discussion if the whole request state validation shouldn't happen in a custom validator, instead of how it's done today.